src/node_options.cc

```diff
@@ -135,9 +135,13 @@ void PerIsolateOptions::HandleMaxOldSpaceSizePercentage(
   // Use uint64_t for the result to prevent data loss on 32-bit systems.
   uint64_t available_memory =
       (constrained_memory > 0 && constrained_memory != UINT64_MAX)
-          ? constrained_memory
+          ? std::min(total_memory, constrained_memory)
```

Comment:
As you correctly noted, `uv_get_total_memory` returns 0 when its value is unknown. This might become problematic if `uv_get_constrained_memory` is defined but `uv_get_total_memory` is not, as the min function would incorrectly return 0 despite a valid constraint being available.